src/Plugin/Field/FieldFormatter/PriceFormatter.php

```diff
@@ -0,0 +1,190 @@
+    parent::__construct($plugin_id, $plugin_definition, $field_definition, $settings, $label, $view_mode, $third_party_settings);
+
+    $this->currencyFormatter = $currency_formatter;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $plugin_id,
+      $plugin_definition,
+      $configuration['field_definition'],
+      $configuration['settings'],
+      $configuration['label'],
+      $configuration['view_mode'],
+      $configuration['third_party_settings'],
+      $container->get('apigee_m10n.price_formatter')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function defaultSettings() {
+    return [
+        'strip_trailing_zeroes' => FALSE,
+        'currency_display' => 'symbol',
+      ] + parent::defaultSettings();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function settingsForm(array $form, FormStateInterface $form_state) {
+    $elements = [];
+    $elements['strip_trailing_zeroes'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Strip trailing zeroes after the decimal point.'),
+      '#default_value' => $this->getSetting('strip_trailing_zeroes'),
+    ];
+    $elements['currency_display'] = [
+      '#type' => 'radios',
+      '#title' => $this->t('Currency display'),
+      '#options' => [
+        'symbol' => $this->t('Symbol (e.g. "$")'),
+        'code' => $this->t('Currency code (e.g. "USD")'),
+        'none' => $this->t('None'),
+      ],
+      '#default_value' => $this->getSetting('currency_display'),
+    ];
+
+    return $elements;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function settingsSummary() {
+    $summary = [];
+    if ($this->getSetting('strip_trailing_zeroes')) {
+      $summary[] = $this->t('Strip trailing zeroes after the decimal point.');
+    }
+    else {
+      $summary[] = $this->t('Do not strip trailing zeroes after the decimal point.');
+    }
+
+    $currency_display = $this->getSetting('currency_display');
+    $currency_display_options = [
+      'symbol' => $this->t('Symbol (e.g. "$")'),
+      'code' => $this->t('Currency code (e.g. "USD")'),
+      'none' => $this->t('None'),
+    ];
+    $summary[] = $this->t('Currency display: @currency_display.', [
+      '@currency_display' => $currency_display_options[$currency_display],
+    ]);
+
+    return $summary;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(FieldItemListInterface $items, $langcode) {
+    $options = $this->getFormattingOptions();
+    $elements = [];
+    foreach ($items as $delta => $item) {
+      $elements[$delta] = [
+        '#markup' => $this->currencyFormatter->format($item->amount, $item->currency_code, $options),
```

Comment:
Should there be a space between the code and the amount? (Note: I changed the first one first one to show currency code)

<img width="300" alt="Screen Shot 2019-06-25 at 12 41 30 PM" src="https://user-images.githubusercontent.com/60979/60116726-9aac5180-9746-11e9-9067-7ffc025d2ac8.png">

Reply:
There are tree ways to configure the formatter and this is what they look like: 

<img width="227" alt="image" src="https://user-images.githubusercontent.com/159693/60146929-4756f500-9780-11e9-83eb-749a43c031de.png">

If there are some improvements to be made, maybe we can address them in follow ups.